app/routes/search.tsx

```diff
@@ -16,16 +16,7 @@ export async function loader({ request }: LoaderFunctionArgs) {
   const user = await auth.isAuthenticated(request)
   const url = new URL(request.url)
 
-  const validParams = ['q', 'attributes_to_crop', 'highlight_pre_tag', 'highlight_post_tag']
-  const query = new URLSearchParams()
-
-  for (const [key, value] of url.searchParams.entries()) {
-    if (validParams.includes(key)) {
-      query.set(key, value)
-    }
-  }
```

Comment:
not a big deal but I think it made more sense to do this conversion to the params object (previous `URLSearchParams`, now a plain object) here, where you can be more confident you're in the context where those params are present on the URL